[espnet3-14] Add integration test

Masao-Someki opened the pull request:
## What did you change?
                                                                                                                             
  - Added an espnet3 integration test in CI and introduced TEMPLATE/mini_an4 ASR example configs, scripts, and dataset       
    generation code.                                                                                                         
  - Added jiwer as a dependency for the scoring stage.                                                                       
  - Moved the mini_an4 dataset archive to egs2 to avoid using egs.                                                           

  ## Why did you make this change?
                                                                                                                             
  - To validate espnet3 end‑to‑end behavior via a reproducible CI integration test.                                          
  - To declare the dependency required for scoring.                                                                          
  - To align with the policy of using egs2 instead of egs.                                                                   

                                                                                                           
  ## Is your PR small enough?                                                                                                
                                                                                                                             
  no                                                                                                                         
  (more than 20 files changed, because it includes TEMPLATE directory.) 
                                                                                             
  ## Additional Context                                                                                                      
                                                                                                                             
  - egs2/mini_an4/asr1/downloads.tar.gz is a binary update.

This PR should be merged after #6329.

* #6133
* #6327
* #6328
* #6329

Masao-Someki committed:
Add integration test

Masao-Someki committed:
Added jiwer for scoring stage

Masao-Someki committed:
Moved mini-an4 dataset from egs to egs2

- This is to avoid using egs folder

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Masao-Someki committed:
Merge branch 'espnet3/package_files' of github.com:Masao-Someki/espnet into espnet3/integration_test

Masao-Someki committed:
Merge branch 'espnet3/integration_test' of github.com:Masao-Someki/espnet into espnet3/integration_test

Masao-Someki committed:
Merge branch 'espnet3/logging_utils' of github.com:Masao-Someki/espnet into espnet3/integration_test

Masao-Someki committed:
Bug fix

- Assume hypothesis to be "" when hypothesis is blank

Masao-Someki committed:
Add validation-based lr scheduler such as ReduceOnPlateau

- Previously we asked developer to create a user-defined modle, but I supported as a default.
- Userd can set `val_scheduler_criterion` as espnet2 to use this function.

Masao-Someki committed:
Skip bulding tokenizer when exists

Masao-Someki committed:
Some bug fix for config loading

- supported train/valid switching for preprocessor
- Add new default resolver to load external config file

Masao-Someki committed:
Add transducer system, task, and inference runner

Masao-Someki committed:
Add configs for integration test

Masao-Someki committed:
Add running script for transducer with Transducer system

Masao-Someki committed:
Add more integration tests for ASR

Masao-Someki committed:
Removed debug line

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Masao-Someki committed:
Format and fixed CI

Masao-Someki committed:
Merge branch 'espnet3/integration_test' of github.com:Masao-Someki/espnet into espnet3/integration_test

Masao-Someki committed:
Merge branch 'espnet3/logging_utils' of github.com:Masao-Someki/espnet into espnet3/integration_test

Masao-Someki committed:
Separate logs for each stages

LiChenda commented on espnet3/systems/asr/metrics/cer.py:
```diff
@@ -0,0 +1,98 @@
+            RuntimeError: If ``jiwer`` is not installed.
+        """
+        if jiwer is None:
+            raise RuntimeError(
```
Oh, here it is. That looks great.

LiChenda reviewed (commented):
I left some simple suggestions. Thank you @Masao-Someki, it's an excellent PR

Masao-Someki committed:
Fixed bug in trainer config

Masao-Someki committed:
Fixed CI

Masao-Someki committed:
Update infer.yaml

Masao-Someki committed:
Fixed configs

Masao-Someki committed:
Merge branch 'master' of https://github.com/espnet/espnet into espnet3/integration_test

Masao-Someki committed:
Merge branch 'espnet3/package_files' of github.com:Masao-Someki/espnet into espnet3/integration_test

Masao-Someki committed:
Bug fixed for the logging directory

- rename decode_dir into infer_dir

Masao-Someki committed:
Removed unrequired files

Masao-Someki committed:
decode-dir to infer-dir

Masao-Someki committed:
removed run_transducer.py from integration test

Masao-Someki committed:
Added a new config infer.yaml: provider.params

I noticed that there is currently no mechanism to configure hyperparameters that control the inference-path.
With this settings users can put hyper parameters for these settings directly from the config file.

Masao-Someki committed:
Fixed add_special_tokens

- Insert special token after the last token if insert_after is not specified.

Masao-Someki committed:
Bug fixed for the instantiating part on inference provider.

Masao-Someki committed:
Merge branch 'master' of https://github.com/espnet/espnet into espnet3/integration_test

Masao-Someki committed:
Merge branch 'espnet3/logging_utils' of github.com:Masao-Someki/espnet into espnet3/integration_test

Masao-Someki committed:
Fixed issues

Masao-Someki committed:
Fixed namings of a mini-an4 recipe

Masao-Someki committed:
Fixed config entries for mini-an4

Masao-Someki committed:
Fixed ci script

Masao-Someki committed:
Fixed integration config

Masao-Someki committed:
measure -> metric

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

sw005320 commented:
@Masao-Someki, please resolve the conflicts and update the branch

Masao-Someki committed:
Merge branch 'espnet3/integration_test' of github.com:Masao-Someki/espnet into espnet3/integration_test

Masao-Someki committed:
Merge branch 'master' of https://github.com/espnet/espnet into espnet3/integration_test

Masao-Someki committed:
Fixed merge mistake

Masao-Someki committed:
Fixed namings

Masao-Someki committed:
Fixed PR scripts

Masao-Someki committed:
Fixed test-integration-espnet3.sh

Masao-Someki committed:
Fixed CI Issue

Fhrozen committed:
Merge branch 'master' into espnet3/integration_test

Masao-Someki committed:
Fixed CI and format infer/measure config

Masao-Someki committed:
Merge branch 'espnet3/integration_test' of github.com:Masao-Someki/espnet into espnet3/integration_test

Masao-Someki committed:
Removed transducer system and fixed unit tests

- Since we have `src/infer.py`, we don't need Transducer system in espnet3.

Masao-Someki committed:
Add tests for CI

sw005320 commented on egs3/mini_an4/asr/src/inference.py:
`egs3/mini_an4/asr/src/inference.py`
-->
`egs3/mini_an4/asr/src/infer.py`

sw005320 reviewed (commented):
They look good in general.

- The config files tend to be very long. Can you try to set some config options to their default values and remove as many lines as possible?
- Please also add comments to the config like the that in the template directory
- what is the difference betwen `egs3/mini_an4/asr/conf/train_asr_transformer.yaml` and `egs3/mini_an4/asr/conf/train_asr_transformer_debug.yaml`?
- Can you remove `_debug` from the yaml file?

Masao-Someki committed:
Bug fix

Masao-Someki committed:
CI fixed

Masao-Someki committed:
Update namings

- train -> training
- file names are nouns
- revised function names to verbs

Masao-Someki committed:
Merge branch 'master' into espnet3/integration_test

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Masao-Someki commented:
I will work on these two points:
>- The config files tend to be very long. Can you try to set some config options to their default values and remove as many lines as possible?
>- Please also add comments to the config like the that in the template directory

I have pushed the change to see if integration test for espnet3 properly works on the new CI workflow

Masao-Someki committed:
Move class paths to TEMPLATE default config.

- move to TEMPLATE/*/*.yaml and load in initialization step

Masao-Someki committed:
Add unit test

Masao-Someki committed:
Applied format

Masao-Someki committed:
Add egs3 as package for default config

Masao-Someki committed:
Comment out unit tests that is fixed in another PR.

Masao-Someki committed:
Enhance config description and move unneccesory configs to optional config.

Masao-Someki committed:
Add comment on measure stage and metrics.yaml

Masao-Someki committed:
Applied format

Masao-Someki committed:
Fixed configs and refactored run.py accordingly

Masao-Someki committed:
Merge branch 'espnet3/integration_test' of github.com:Masao-Someki/espnet into espnet3/integration_test

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Masao-Someki committed:
Add comments on config, reflect review

Masao-Someki committed:
Bug fixed

Masao-Someki committed:
Merge branch 'espnet3/integration_test' of github.com:Masao-Someki/espnet into espnet3/integration_test

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Masao-Someki committed:
Merge branch 'master' into espnet3/integration_test

Masao-Someki committed:
Merge branch 'espnet3/integration_test' of github.com:Masao-Someki/espnet into espnet3/integration_test

Masao-Someki committed:
Fixed readme

Masao-Someki committed:
train_config -> training_config, infer_config -> inference_config, measure_config -> metrics_config

Masao-Someki committed:
Moved recipe-specific code to run.py

Masao-Someki committed:
Fixed CI bug

Masao-Someki committed:
Fixed CI bug in integration test

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Masao-Someki committed:
Fixed CI issue

Masao-Someki committed:
Merge branch 'espnet3/integration_test' of github.com:Masao-Someki/espnet into espnet3/integration_test

Masao-Someki committed:
Fixed CI issue

Masao-Someki committed:
Merge branch 'master' into espnet3/integration_test

Masao-Someki commented:
I will close this PR as this PR has been merged via #6390, #6391, #6392, #6416
